### DISCUSSION TITLE
License

### DISCUSSION THREAD

**Discussion by mzayats** [Skylar-Tech/node-red-contrib-matrix-chat#113] [General]
If I use these nodes in my own flows and embed them in a broader app, would I have to release the entire app under GPLv3?

**Comment by skylord123** [comment]
https://fossa.com/blog/open-source-software-licenses-101-gpl-v3/

> Using the Licensed Code
> The GPL v3 license permits users of the code to:
> 
> **Use the code for commercial purposes:** 
> Like GPL v2, GPL v3 imposes no conditions on the internal use of the software.
> **Change the code:**
> Users can change or rework the code, but if they distribute these changes/modifications in binary form, they’re also required to release these updates in source code form under the GPL v3 license.
> **Distribute copies or modifications of the code:**
> As long as these modifications are also released under the GPL v3 license, they can be distributed to others.
> **Place warranty:**
> Distributors of the original code can offer their own warranty on the licensed software.

You are fine to use the software and even package and distribute it as long as you don't make changes to this code or if you do you need to open-source it or merge the changes back up so others can benefit. 

The idea is to keep someone from coming along and taking the code, making improvements (to the source code, not by using them in flows), then keeping that private source and distributing it to people in their own software. Any changes made to this module should be open for all.

Thanks,
Skylar
